File: src/main/java/com/uwetrottmann/trakt5/entities/PersonIds.java
/*
 * Copyright 2024 Uwe Trottmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.uwetrottmann.trakt5.entities;

import javax.annotation.Nonnull;

public class PersonIds extends BaseIds {

    public String slug;
    /**
     * @deprecated This appears to be no longer supported.
     */
    @Deprecated
    public String tvrage;

    @Nonnull
    public static PersonIds trakt(int trakt) {
        PersonIds ids = new PersonIds();
        ids.trakt = trakt;
        return ids;
    }

    @Nonnull
    public static PersonIds imdb(String imdb) {
        PersonIds ids = new PersonIds();
        ids.imdb = imdb;
        return ids;
    }

    @Nonnull
    public static PersonIds tmdb(int tmdb) {
        PersonIds ids = new PersonIds();
        ids.tmdb = tmdb;
        return ids;
    }

    @Nonnull
    public static PersonIds slug(String slug) {
        PersonIds ids = new PersonIds();
        ids.slug = slug;
        return ids;
    }
}

File: src/main/kotlin/com/redissi/trakt/entities/PersonIds.kt
package com.redissi.trakt.entities

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class PersonIds(
    override val trakt: Int? = null,
    override val imdb: String? = null,
    override val tmdb: Int? = null,
    val slug: String? = null,
    val tvrage: String? = null
) : BaseIds() {

    companion object {
        fun trakt(trakt: Int): PersonIds {
            return PersonIds(trakt = trakt)
        }

        fun imdb(imdb: String?): PersonIds {
            return PersonIds(imdb = imdb)
        }

        fun tmdb(tmdb: Int): PersonIds {
            return PersonIds(tmdb = tmdb)
        }

        fun slug(slug: String?): PersonIds {
            return PersonIds(slug = slug)
        }
    }
}